lib/helpers/configuration.rb

```diff
@@ -20,10 +20,11 @@ def configuration
     yield self
   end
 
-  def define_setting(name, default = nil)
+  def define_setting(name, default = nil, allow_nil: true)
     class_variable_set("@@#{name}", default)
 
     define_class_method "#{name}=" do |value|
+      raise "#{name} cannot be nil." if value.nil? && !allow_nil
```

Comment:
LOVE IT! :heart: 

Couple of suggestions:
- Since we don't have any config settings that can be nil. I think you could remove `allow_nil`.

- For the error message, how about. `"ShareMeowClient.#{name} was not set. Please set it in your ShareMeow initializer"`